frontend/src/choreoComponents/ChoreoDetail.tsx

```diff
@@ -4,62 +4,278 @@ import {
+                                <Grid container spacing={2} justifyContent="space-evenly" mt={2} mb={2}>
+                                    <Button
+                                        id="galleryBtn"
+                                        type="button"
+                                        variant="contained"
+                                        onClick={() => {
+                                            navigate("/choreo")
+                                        }}
+                                        startIcon={<ArrowBackIcon/>}>
+                                        Back to gallery
+                                    </Button>
+                                    <Button
+                                        id="galleryBtn"
+                                        variant="contained"
+                                        color="success"
+                                        startIcon={<AddIcon/>}
+                                        onClick={handleOpen}>
+                                        Add Move
+                                    </Button>
+                                    <Button
+                                        id="galleryBtn"
+                                        type="submit"
+                                        variant="contained"
+                                        startIcon={<SaveIcon/>}
+                                    >
+                                        Save Update
+                                    </Button>
+                                    <Button
+                                        id="deleteBtn"
+                                        type="submit"
+                                        variant="outlined"
+                                        color="error"
+                                        startIcon={<DeleteIcon/>}
+                                        onClick={deleteChoreoOnClick}
+                                    >
+                                        Delete Choreo
+                                    </Button>
+                                </Grid>
+                            </Grid>
+                        </form>
+                    </Container>
+                    <Dialog open={open} onClose={handleClose}>
+                        <DialogTitle style={{backgroundColor: "lightgrey", color: "#1B1E24"}}>
+                            Click on move to add
+                        </DialogTitle>
+                        <DialogContent style={{backgroundColor: "lightgrey"}}>
+                            <DialogContentText>
+                                <ChoreoMoveGallery moves={props.moves} onSelectMove={handleSelectMove}/>
+                            </DialogContentText>
+                        </DialogContent>
+                        <DialogActions style={{backgroundColor: "lightgrey"}}>
+                            <Button onClick={handleClose}>Cancel</Button>
+                        </DialogActions>
```

Comment:
Optional: du könntest hier den Dialog noch in eine Komponente auslagern, das würde die DetailsComponent etwas kürzer machen